src/main/java/org/elasticsearch/rest/action/admin/indices/analyze/RestAnalyzeAction.java

```diff
@@ -62,6 +64,74 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
+            } else {
+                XContentParser.Token token;
+                String currentFieldName = null;
+                int count = 0;
+                while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+                    if (token == XContentParser.Token.FIELD_NAME) {
+                        currentFieldName = parser.currentName();
+                    } else if ("prefer_local".equals(currentFieldName) && token == XContentParser.Token.VALUE_BOOLEAN) {
+                        analyzeRequest.preferLocal(parser.booleanValue());
+                    } else if ("text".equals(currentFieldName) && token == XContentParser.Token.VALUE_STRING) {
+                            analyzeRequest.text(parser.text());
+                    } else if ("analyzer".equals(currentFieldName) && token == XContentParser.Token.VALUE_STRING) {
+                        analyzeRequest.analyzer(parser.text());
+                    } else if ("field".equals(currentFieldName) && token == XContentParser.Token.VALUE_STRING) {
+                        analyzeRequest.field(parser.text());
+                    } else if ("tokenizer".equals(currentFieldName) && token == XContentParser.Token.VALUE_STRING) {
+                        analyzeRequest.tokenizer(parser.text());
+                    }else if (("token_filters".equals(currentFieldName) || "filters".equals(currentFieldName)) && token == XContentParser.Token.START_ARRAY) {
+                        List<String> filters = Lists.newArrayList();
+                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                            if (token.isValue() == false) {
+                                throw new ElasticsearchIllegalArgumentException(currentFieldName + " array element should only contain token filter's name");
+                            }
+                            filters.add(parser.text());
+                        }
+                        analyzeRequest.tokenFilters(filters.toArray(new String[0]));
+                    } else if ("char_filters".equals(currentFieldName) && token == XContentParser.Token.START_ARRAY) {
+                        List<String> charFilters = Lists.newArrayList();
+                        while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                            if (token.isValue() == false) {
+                                throw new ElasticsearchIllegalArgumentException(currentFieldName + " array element should only contain char filter's name");
+                            }
+                            charFilters.add(parser.text());
+                        }
+                        analyzeRequest.tokenFilters(charFilters.toArray(new String[0]));
+                    } else {
+                        throw new ElasticsearchIllegalArgumentException("Unknown param [" + currentFieldName + "] in request body");
```

Comment:
I think you should added or "parameter is of the wrong type" since it may be a valid name ..